src/arch/xtensa/include/arch/task.h

```diff
@@ -159,13 +160,21 @@ static void _irq_task(void *arg)
 		task = container_of(clist, struct task, irq_list);
 		list_item_del(clist);
 
+		if (task->func && task->state == TASK_STATE_PENDING) {
```

Comment:
Maybe we should not check state here. In case previous task complete have changed the state to COMPLETED to make us drop ipc.

Reply:
agree, let's change this line to 
```
if (task->func) {
```
for hot fix.